creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -274,15 +293,75 @@ class SnapbackSM {
   }
 
   /**
-   * Submit POA:UserReplicaSetFactory.updateReplicaSet() transaction to move user off unhealthy node
+   * Depending on the size of `unhealthyReplicas`:
+   * 1. Determine a new replica set
+   * 2. Write new replica set to URSM
+   * 3. Sync data to new replica set
    *
-   * @dev TODO use libs for CreatorNodeSelection logic + relayed chain call
-   * @dev One issue - since it doesn't actually rectify broken replica sets, this will log every time it is processed
-   */
-  async issueUpdateReplicaSetOp (userId, wallet, primary, secondary1, secondary2, unhealthyReplica) {
-    // await libs.userReplicaSetManagerClient._updateReplicaSet(userId, primary, [new secondaries], primary, [old secondaries])
+   * @param {number} userId user id to issue a reconfiguration for
+   * @param {string} wallet wallet address of user id
+   * @param {string} primary endpoint of the current primary node on replica set
+   * @param {string} secondary1 endpoint of the current first secondary node on replica set
+   * @param {string} secondary2 endpoint of the current second secondary node on replica set
+   * @param {string[]} unhealthyReplicas array of endpoints of current replica set nodes that are unhealthy
+   * @param {string[]} healthyPeers array of endpoints of current peer set nodes slice that are healthy
+  */
+  async issueUpdateReplicaSetOp (userId, wallet, primary, secondary1, secondary2, unhealthyReplicas, healthyPeers) {
+    this.log(`[issueUpdateReplicaSetOp] userId=${userId} wallet=${wallet} unhealthy replica set=[${unhealthyReplicas}] | current healthy peer set=[${healthyPeers}]`)
+
+    // TODO: make optimizely flag
+    if (!this.snapbackReconfigEnabled) return
```

Comment:
p sure in order to wire this up to optimizely, it needs to be a libs param?

Reply:
ah yes. I can set it up so if you or I eventually make this an optimizely flag, it'll be straight forward. updating

Reply:
perf